src/reactor.cc

```diff
@@ -0,0 +1,406 @@
+      ->AddVal("TritiumCore", core)
+      ->AddVal("TritiumReserve", reserve)
+      ->AddVal("TritiumStorage", storage)
+      ->AddVal("LithiumBlanket", blanket)
+      ->AddVal("HeliumStorage", helium)
+      ->Record();
+}
+
+void Reactor::DepleteBlanket(double bred_tritium_mass) {
+  cyclus::Material::Ptr blanket_mat = blanket.Pop();
+
+  cyclus::CompMap b = blanket_mat->comp()->mass();
+  cyclus::compmath::Normalize(&b, blanket_mat->quantity());
+
+  // Percent (as decimal) of T which comes from Li-7 instead of Li-6
+  double Li7_contribution = 0.15;
+
+  cyclus::CompMap depleted_comp;
+
+  // This is ALMOST the correct behavior, but "scraping the bottom of the
+  // barrel" is a little too complex for this implementation.
+  if ((b[30060000] - (1 - Li7_contribution) * 2 * bred_tritium_mass > 0) &&
+      (b[30070000] - Li7_contribution * 7.0 / 3.0 * bred_tritium_mass > 0)) {
+    depleted_comp = {{30070000, b[30070000] - Li7_contribution * 7.0 / 3.0 *
+                                                  bred_tritium_mass},
+                     {30060000, b[30060000] - (1 - Li7_contribution) * 2 *
+                                                  bred_tritium_mass},
+                     {10030000, b[10030000] + bred_tritium_mass},
+                     {20040000, b[20040000] + 4.0 / 3.0 * bred_tritium_mass}};
+
+    // Account for the added mass of the absorbed neutrons
+    double neutron_mass_correction =
+        1.0 / 3.0 * bred_tritium_mass * (1 - Li7_contribution);
+    cyclus::Material::Ptr additional_mass = cyclus::Material::Create(
+        this, neutron_mass_correction,
+        cyclus::Composition::CreateFromMass(depleted_comp));
+
+    blanket_mat->Transmute(cyclus::Composition::CreateFromMass(depleted_comp));
+    blanket_mat->Absorb(additional_mass);
+
+    RecordOperationalInfo("Blanket Depletion",
+                          "Tritium bred at perscribed rate");
+  } else {
+    RecordOperationalInfo(
+        "Breeding Error",
+        "Blanket composition lacks sufficient lithium to continue "
+        "breeding at perscribed rate");
+  }
+  blanket.Push(blanket_mat);
+}
+
+cyclus::Material::Ptr Reactor::BreedTritium(double fuel_usage, double TBR) {
+  DepleteBlanket(fuel_usage * TBR);
+  cyclus::Material::Ptr mat = blanket.Pop();
+
+  cyclus::CompMap c = mat->comp()->mass();
+  cyclus::compmath::Normalize(&c, mat->quantity());
+
+  cyclus::CompMap T = {{10030000, 1}};
+
+  cyclus::Material::Ptr bred_fuel =
+      mat->ExtractComp(c[10030000], cyclus::Composition::CreateFromAtom(T));
+  blanket.Push(mat);
+
+  RecordOperationalInfo("Bred Tritium", std::to_string(bred_fuel->quantity()) +
+                                            " kg of T bred from blanket");
+
+  return bred_fuel;
+}
+
+void Reactor::OperateReactor(double TBR, double burn_rate) {
+  int seconds_per_year = 31536000;
+  double fuel_usage =
+      burn_rate * (fusion_power / 1000) / seconds_per_year * context()->dt();
```

Comment:
Magic number: what is 1000 here? Assign it to a variable (I assume its MW2kW)

Reply:
First two points addressed, last point will be addressed shortly when I make the big kg --> atoms change and make everything more precise.

Reply:
Decided to not convert burn_rate to atoms, just because I found a different way of doing it which made more sense to me, but which I think achieves the same result. This may have been "the hard way", who knows?